backend/btrixcloud/migrations/migration_0039_coll_slugs.py

```diff
@@ -16,23 +21,69 @@ class Migration(BaseMigration):
     def __init__(self, mdb, **kwargs):
         super().__init__(mdb, migration_version=MIGRATION_VERSION)
 
+    async def dedup_slug(
+        self, name: str, slug_base: str, coll_id: UUID, colls_mdb
+    ) -> None:
+        """attempt to set slug, if duplicate, append suffix until a valid slug is found
+        also update original name with same suffix"""
+        slug = slug_base
+        count = 1
+
+        while True:
+            try:
+                await colls_mdb.find_one_and_update(
+                    {"_id": coll_id},
+                    {"$set": {"slug": slug}},
+                )
+                print("Slug set: " + slug)
+                break
+            except DuplicateKeyError:
+                # pylint: disable=raise-missing-from
+                count += 1
+                slug = f"{slug_base}-{count}"
+
+        if count > 1:
+            print(f"Dedup name '{name}' -> '{name} {count}'")
```

Comment:
Again useful for testing, not sure if we want to keep for prod. If we do we should change the message to be more clear what this applies to

Reply:
Hm, i think it might be useful, since it changes data, and is only printed once?

Reply:
I've done things like this in the past in migrations and switched to only logging errors when I found in prod we otherwise printed a huge number of lines that made tailing the backend logs to make sure the migrations didn't throw any errors more difficult.

If we're going to keep it, i'd add some context like: "Duplicate collection {name} slug set to {name}-{count}" or something so that it makes more sense in context

Reply:
This would I'm more inclined to keep than the other because it represents a sort of exception, but then in that case I'd modify the message to be more understandable at a glance, like:

```
print(f"Duplicate collection name '{name}' slug set to '{name}-{count}'")
```

Otherwise you kinda have to know what that particular migration does to understand it